### ISSUE TITLE
Question: Performance benefit of source location disabling

### ISSUE THREAD

**Issue by pmouawad** [phax/ph-css#54]
Hello,
First thanks for your library and constant improvements.
At JMeter I upgraded yesterday to last version of the library from :
ph-commons.version=9.2.1 => 9.3.7
ph-css.version=6.1.1 => 6.2.0

https://github.com/apache/jmeter/commit/58b571e33925989d5d1920d50416e863b12c36f3

Following that, our TestCSSParser.java:
https://github.com/apache/jmeter/blob/master/src/protocol/http/src/test/java/org/apache/jmeter/protocol/http/parser/TestCssParser.java
failed because it seems you don't throw anymore an exception with Browser Compliant mode set to true.
The CSS we were using was:

`@-webkit-keyframes "introjspulse"{
	0%{-webkit-transform:scale(0);opacity:.0}
	25%{-webkit-transform:scale(0);opacity:.1}
    50%{-webkit-transform:scale(0.1);opacity:.3}
    75%{-webkit-transform:scale(0.5);opacity:.5}
    100%{-webkit-transform:scale(1);opacity:.0}
}`

I think it's related to fix of https://github.com/phax/ph-css/issues/41 and some strengthening on parser:
https://github.com/apache/jmeter/commit/926f895bfdc53f60b9d5b3e77ef231ff15d3b1e6#diff-62a95df1b5df1811a9f469801ffef388

1/ Is my hypothesis correct ?
2/ In your release notes, you mention performance impact of using setUseSourceLocation(true). Did you quantify the negative impact ?
Our current config is done here:

https://github.com/apache/jmeter/blob/master/src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/parser/CssParserCacheLoader.java

Thank you

**Comment by phax** [comment]
Dear Philippe,

ad 1/ yes your assumption is correct - due to an improvement in error handling based on #41, ph-css 6.2.0 throws less exceptions on errors, as long as the "browser compliant mode" is enabled.

ad 2/ No, unfortunately I never measured it. But basically three new object are allocated for each domain objects which ends up in quite some objects. I did a very quick test on my test class `CSSReader30FuncTest` - I did a couple of runs (after warmup) and the difference was 14 seconds (with sourcelocation off) to 18.5 seconds (with sourcelocation on), so I think it DOES have a measurable impact.

I looked at you current configuration and it looks totally fine - no input from my side :)
BR, Philip
